.github/workflows/ci_pr.yml

```diff
@@ -0,0 +1,140 @@
+            PYLINTOPTS: "--rcfile=ci/2.7.pylintrc"
+            NOSEOPTS: "--verbose --with-timer"
+            PYLINTFILES: "azurelinuxagent setup.py makepkg.py tests"
+
+          - python-version: 3.4
+            PYLINTOPTS: "--rcfile=ci/2.7.pylintrc"
+            NOSEOPTS: "--verbose --with-timer"
+            PYLINTFILES: "azurelinuxagent setup.py makepkg.py tests"
+
+          - python-version: 3.6
+            PYLINTOPTS: "--rcfile=ci/3.6.pylintrc"
+            NOSEOPTS: "--verbose --with-timer"
+            PYLINTFILES: "azurelinuxagent setup.py makepkg.py tests"
+
+          - python-version: 3.7
+            PYLINTOPTS: "--rcfile=ci/3.6.pylintrc"
+            NOSEOPTS: "--verbose --with-timer"
+            PYLINTFILES: "azurelinuxagent setup.py makepkg.py tests"
+
+          - python-version: 3.8
+            PYLINTOPTS: "--rcfile=ci/3.6.pylintrc"
+            NOSEOPTS: "--verbose --with-coverage --with-xunit --xunit-file=nosetests.__uuid__.xml
+              --with-timer --cover-erase --cover-inclusive --cover-branches --cover-package=azurelinuxagent"
+            PYLINTFILES: "azurelinuxagent setup.py makepkg.py tests"
+    
+    name: "Python ${{ matrix.python-version }} Unit Tests"
+    runs-on: ubuntu-18.04
+
+    env:
+      PYLINTOPTS: ${{ matrix.PYLINTOPTS }}
+      PYLINTFILES: ${{ matrix.PYLINTFILES }}
+      NOSEOPTS: ${{ matrix.NOSEOPTS }}
+      PYTHON_VERSION: ${{ matrix.python-version }}
+
+    steps:
+    
+    - name: Checkout WALinuxAgent repo
+      uses: actions/checkout@v2
+
+    - name: Setup Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    
+    - name: Install dependencies
+      run: |
+        sudo env "PATH=$PATH" python -m pip install --upgrade pip
+        sudo env "PATH=$PATH" pip install -r requirements.txt
+        sudo env "PATH=$PATH" pip install -r test-requirements.txt
+
+    - name: Run pylint
+      run: |
+        pylint $PYLINTOPTS --jobs=0 $PYLINTFILES
+
+    - name: Test with nosetests
+      if: always()
+      run: |
+        ./ci/nosetests.sh || EXIT_CODE=$?
+        exit $EXIT_CODE
+
+    - name: Run Coverage
+      env:
+        MIN_COVERAGE_GOAL: 60
+      if: matrix.python-version == 3.8
+      run: |
+        echo looking for coverage files :
+        ls -alh | grep -i coverage
+        sudo env "PATH=$PATH" coverage combine coverage.*.data
+        sudo env "PATH=$PATH" coverage xml
+        COVERAGE_PERCENT=$(coverage report | tail -n 1 | rev | cut -d ' ' -f 1 | cut -c 2- | rev)
+        EXIT_CODE=1
+        echo Codecoverage : $COVERAGE_PERCENT%
+        if [ "$COVERAGE_PERCENT" -eq "$COVERAGE_PERCENT" ] && [ "$COVERAGE_PERCENT" -gt "$MIN_COVERAGE_GOAL" ]; then 
+          echo Pass coverage check.
+          EXIT_CODE=0;
+        else 
+          echo Failed coverage check. Found $COVERAGE_PERCENT%, expected minimum of $MIN_COVERAGE_GOAL%;
+          EXIT_CODE=1;
```

Comment:
So far we've not had `codecov` as a required feature, with this though we would enforce min 60% right? I wonder if we should make this an enforcing thing or not. Maybe just publish the coverage data but dont fail if coverage < 60%?

Reply:
The previous travis script had code coverage silently enforced to 60%
If the coverage (I think total coverage, not PR coverage...) dropped under 60%, the PR would have failed.

Reply:
The PR wouldn't have failed. The coverage step would fail, but we didn't treat it as a required check for merging.

I think we should have coverage as an optional check, at least in the beginning.

Reply:
If we make code cov a separate job, that would be easy to configure via the repo's settings page.

Reply:
@pgombar It woudnt have failed because there was a bug in the previous script :
 NOSEOPTS were not getting passed to any run, including python 3.8

If the argument were passed correctly, python3.8 would run coverage validation during unit test, and would fail the unit test part if its sub 60%

Reply:
No, I'm talking about the time when we were actually running codecov, before the bug. The codecov check fails, but the PR doesn't fail because we set the codecov hook as optional in the settings for develop branch.

Reply:
I've kept the codecov upload, but removed the check for >60% coverage. That way, we can get the coverage info from the link, but it doesn't fail the workflow, and we can still merge a PR regardless.

This is going back to the stated purpose of this PR, which was to replicate the functionality we had with Travis CI in Github Actions.